### PR TITLE
[hi] Add content/hi/docs/setup/production-environment/container-runtimes.md

### PR DESCRIPTION
**Pull request by Darshnadas**
This PR is a part of ongoing hi localization

### CODE REVIEW

File: content/hi/docs/setup/production-environment/container-runtimes.md [hunk 1]
```diff
@@ -0,0 +1,388 @@
...
+
+
+Systemd और cgroups एकीकृत हैं और प्रत्येक systemd यूनिट को एक cgroup आवंटित होता है। अपने कन्टैनर रनटाइम और kubelet को `cgroupfs` प्रयोग करने के लिए कॉन्फ़िगर करना संभव है। systemd के साथ `cgroupfs` प्रयोग करने के कारण दो अलग cgroup मैनेजर होंगे।
+
+एक एकल cgroup प्रबंधक इस दृष्टिकोण को सरल बनाता है कि कौन से संसाधन आवंटित किए जा रहे हैं और डिफ़ॉल्ट रूप से उपलब्ध और उपयोग में आने वाले संसाधनों के बारे में अधिक सुसंगत दृश्य होगा।
+जब एक सिस्टम पर दो cgroup मैनेजर होते हैं, तो आपको उन रिसोर्सेज के दो व्यू मिलते हैं। क्षेत्र में, लोगों ने ऐसे मामलों की सूचना दी है जहां नोड्स जो kubelet और डॉकर के लिए `cgroupfs` का उपयोग करने के लिए कॉन्फ़िगर किए गए हैं, लेकिन बाकी प्रक्रियाओं के लिए `systemd` संसाधन दबाव के कारण अस्थिर हो जाते हैं। 
+
+सेटिंग्स को इस तरह बदलना कि आपका कंटेनर रनटाइम और kubelet `systemd` का उपयोग करें क्योंकि cgroup ड्राइवर सिस्टम को स्थिर करता है। डॉकर के लिए इसे कॉन्फ़िगर करने के लिए, `native.cgroupdriver=systemd` सेट करें। 
+
+{{<caution >}}
```
Comment (anubha-v-ardhan) [inline content/hi/docs/setup/production-environment/container-runtimes.md:40]:
```suggestion
{{< caution >}}
```

**Review by anubha-v-ardhan — CHANGES_REQUESTED** [review]

File: content/hi/docs/setup/production-environment/container-runtimes.md [hunk 2]
```diff
@@ -0,0 +1,388 @@
+title: कंटेनर रनटाइम
+content_type: concept
+weight: 20
+---
```
Comment (anubha-v-ardhan) [inline content/hi/docs/setup/production-environment/container-runtimes.md:4]:
```suggestion
---
title: कंटेनर रनटाइम
content_type: concept
weight: 20
---
```
?

File: content/hi/docs/setup/production-environment/container-runtimes.md [hunk 3]
```diff
@@ -0,0 +1,388 @@
...
+<!-- overview -->
+
+आपको क्लस्टर में प्रत्येक नोड में एक {{< glossary_tooltip text="container रनटाइम" term_id="container-runtime" >}}
```
Comment (anubha-v-ardhan) [inline content/hi/docs/setup/production-environment/container-runtimes.md:7]:
```suggestion
आपको क्लस्टर में प्रत्येक नोड में एक {{< glossary_tooltip text="कंटेनर रनटाइम" term_id="container-runtime" >}} 
```
?

File: content/hi/docs/setup/production-environment/container-runtimes.md [hunk 4]
```diff
@@ -0,0 +1,388 @@
...
+
+- [containerd](#containerd)
+- [CRI-O](#cri-o)
+- [Docker](#docker)
+
+{{< note >}}
+अन्य ऑपरेटिंग सिस्टम के लिए, अपने प्लेटफ़ॉर्म के लिए अनुसार विशिष्ट प्रलेखन देखें।
+{{< /note >}}
+
+## cgroup ड्राइवर
```
Comment (anubha-v-ardhan) [inline content/hi/docs/setup/production-environment/container-runtimes.md:24]:
```suggestion
## Cgroup ड्राइवर 
```

File: content/hi/docs/setup/production-environment/container-runtimes.md [hunk 5]
```diff
@@ -0,0 +1,388 @@
...
+जब एक सिस्टम पर दो cgroup मैनेजर होते हैं, तो आपको उन रिसोर्सेज के दो व्यू मिलते हैं। क्षेत्र में, लोगों ने ऐसे मामलों की सूचना दी है जहां नोड्स जो kubelet और डॉकर के लिए `cgroupfs` का उपयोग करने के लिए कॉन्फ़िगर किए गए हैं, लेकिन बाकी प्रक्रियाओं के लिए `systemd` संसाधन दबाव के कारण अस्थिर हो जाते हैं। 
+
+सेटिंग्स को इस तरह बदलना कि आपका कंटेनर रनटाइम और kubelet `systemd` का उपयोग करें क्योंकि cgroup ड्राइवर सिस्टम को स्थिर करता है। डॉकर के लिए इसे कॉन्फ़िगर करने के लिए, `native.cgroupdriver=systemd` सेट करें। 
+
+{{< /caution >}}
+क्लस्टर में शामिल होने वाले नोड के cgroup ड्राइवर को बदलना एक संवेदनशील ऑपरेशन है। यदि kubelet ने एक सीग्रुप ड्राइवर के शब्दार्थ का उपयोग करके पॉड्स बनाए हैं, तो कंटेनर रनटाइम को दूसरे सीग्रुप ड्राइवर में बदलने से मौजूदा पॉड्स के पॉड सैंडबॉक्स को फिर से बनाते समय त्रुटियां हो सकती हैं। kubelet को पुनरारंभ करने से ऐसी त्रुटियों का समाधान नहीं हो सकता है। 
+यदि आपके पास स्वचालन है जो इसे व्यवहार्य बनाता है, तो अद्यतन किए गए कॉन्फ़िगरेशन का उपयोग करके नोड को दूसरे के साथ बदलें, या स्वचालन का उपयोग करके इसे पुनर्स्थापित करें।
+
```
Comment (anubha-v-ardhan) [inline content/hi/docs/setup/production-environment/container-runtimes.md:43]:
```suggestion
```

File: content/hi/docs/setup/production-environment/container-runtimes.md [hunk 6]
```diff
@@ -0,0 +1,388 @@
...
+{{< /caution >}}
+
+## cgroup संस्करण 2 {#cgroup-v2}
+
+Cgroup v2, cgroup Linux API का अगला संस्करण है।cgroup v1 से अलग, प्रत्येक कंट्रोलर के लिए एक अलग अनुक्रम के बजाय एक एकल पदानुक्रम है।
```
Comment (anubha-v-ardhan) [inline content/hi/docs/setup/production-environment/container-runtimes.md:48]:
```suggestion
Cgroup v2, cgroup Linux API का अगला संस्करण है। cgroup v1 से अलग, प्रत्येक कंट्रोलर के लिए एक अलग अनुक्रम के बजाय एक एकल पदानुक्रम है। 
```

File: content/hi/docs/setup/production-environment/container-runtimes.md [hunk 7]
```diff
@@ -0,0 +1,388 @@
...
+## cgroup संस्करण 2 {#cgroup-v2}
+
+Cgroup v2, cgroup Linux API का अगला संस्करण है।cgroup v1 से अलग, प्रत्येक कंट्रोलर के लिए एक अलग अनुक्रम के बजाय एक एकल पदानुक्रम है। 
+
+नया संस्करण cgroup v1 पर कई सुधार प्रदान करता है, इनमें से कुछ सुधार हैं:
+- API का उपयोग करने का क्लीनर और आसान तरीका
+- कंटेनरों के लिए सुरक्षित उप-वृक्ष प्रतिनिधिमंडल
+- प्रेशर स्टॉल की जानकारी जैसी नई सुविधाएँ
+
+भले ही कर्नेल हाइब्रिड कॉन्फ़िगरेशन का समर्थन करता हो, जहां कुछ नियंत्रक cgroup v1 . द्वारा प्रबंधित किए जाते हैं और कुछ अन्य cgroup v2 द्वारा, Kubernetes सभी नियंत्रकों को प्रबंधित करने के लिए केवल उसी cgroup संस्करण का समर्थन करता है।
```
Comment (anubha-v-ardhan) [inline content/hi/docs/setup/production-environment/container-runtimes.md:55]:
```suggestion
भले ही कर्नेल हाइब्रिड कॉन्फ़िगरेशन का समर्थन करता हो, जहां कुछ नियंत्रक cgroup v1 द्वारा प्रबंधित किए जाते हैं और कुछ अन्य cgroup v2 द्वारा, Kubernetes सभी नियंत्रकों को प्रबंधित करने के लिए केवल उसी cgroup संस्करण का समर्थन करता है।
```
?

File: content/hi/docs/setup/production-environment/container-runtimes.md [hunk 8]
```diff
@@ -0,0 +1,388 @@
...
+
+cgroup v2 में स्विच करते समय उपयोगकर्ता अनुभव में कोई उल्लेखनीय अंतर नहीं होना चाहिए, जब तक कि उपयोगकर्ता सीग्रुप फाइल सिस्टम को सीधे नोड पर या कंटेनरों के भीतर से एक्सेस कर रहे हैं। 
+
+इसका उपयोग करने के लिए, cgroup v2 को CRI रनटाइम द्वारा भी समर्थित होना चाहिए। 
+
+### Kubeadm प्रबंधित क्लस्टर में `systemd` ड्राइवर में माइग्रेट करना
+
+यदि आप मौजूदा kubeadm प्रबंधित क्लस्टर में `systemd` cgroup ड्राइवर में माइग्रेट करना चाहते हैं, तो [माइग्रेशन गाइड](/docs/tasks/administer-cluster/kubeadm/configure-cgroup-driver/) का पालन करें।
+
+## Container runtimes
```
Comment (anubha-v-ardhan) [inline content/hi/docs/setup/production-environment/container-runtimes.md:75]:
```suggestion
## कंटेनर रनटाइम 
```

File: content/hi/docs/setup/production-environment/container-runtimes.md [hunk 9]
```diff
@@ -0,0 +1,388 @@
...
+cat <<EOF | sudo tee /etc/sysctl.d/99-kubernetes-cri.conf
+net.bridge.bridge-nf-call-iptables  = 1
+net.ipv4.ip_forward                 = 1
+net.bridge.bridge-nf-call-ip6tables = 1
+EOF
+# Apply sysctl params without reboot
+sudo sysctl --system
+```
+
+कंटेनर स्थापित करें:
```
Comment (anubha-v-ardhan) [inline content/hi/docs/setup/production-environment/container-runtimes.md:102]:
```suggestion
Containerd इंस्टॉल करें:
```

Comment (anubha-v-ardhan) [reply content/hi/docs/setup/production-environment/container-runtimes.md:102]: container and Containerd are different things

File: content/hi/docs/setup/production-environment/container-runtimes.md [hunk 10]
```diff
@@ -0,0 +1,388 @@
...
+
+   ```powershell
+   .\containerd.exe --register-service
+   Start-Service containerd
+   ```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+#### `systemd` cgroup ड्राइवर {#containerd-systemd} का उपयोग करें
```
Comment (anubha-v-ardhan) [inline content/hi/docs/setup/production-environment/container-runtimes.md:160]:
```suggestion
#### `systemd` cgroup ड्राइवर का उपयोग करें {#containerd-systemd}
```
?

File: content/hi/docs/setup/production-environment/container-runtimes.md [hunk 11]
```diff
@@ -0,0 +1,388 @@
...
+```
+
+यदि आप इस परिवर्तन को लागू करते हैं, तो सुनिश्चित करें कि आप फिर से कंटेन्डर को पुनरारंभ करें: 
+
+```shell
+sudo systemctl restart containerd
+```
+
+जब भी आप kubeadm का उपयोग करें, मैन्युअल रूप से कॉन्फ़िगर करें !
+[क्यूबलेट के लिए cgroup ड्राइवर](/docs/setup/production-environment/tools/kubeadm/install-kubeadm/#configure-cgroup-driver-used-by-kubelet-on-control-plane-node).
```
Comment (anubha-v-ardhan) [inline content/hi/docs/setup/production-environment/container-runtimes.md:179]:
```suggestion
जब आप kubeadm का उपयोग करें, मैन्युअल रूप से कॉन्फ़िगर करें
[क्यूबलेट के लिए cgroup ड्राइवर](/docs/setup/production-environment/tools/kubeadm/install-kubeadm/#configure-cgroup-driver-used-by-kubelet-on-control-plane-node) मैन्युअल रूप से कॉन्फ़िगर करें।
```

File: content/hi/docs/setup/production-environment/container-runtimes.md [hunk 12]
```diff
@@ -0,0 +1,388 @@
...
+
+### CRI-O
+
+इस खंड में CRI-O को कंटेनर रनटाइम के रूप में स्थापित करने के लिए आवश्यक जानकारी है।
+
+अपने सिस्टम पर CRI-O स्थापित करने के लिए निम्नलिखित कमांड का उपयोग करें: 
+
+{{< note >}}
+CRI-O के प्रमुख और छोटे संस्करणों को Kubernetes के प्रमुख और छोटे संस्करणों से मेल खाना चाहिए। 
+अधिक जानकारी के लिए देखें  [CRI-O compatibility matrix](https://github.com/cri-o/cri-o#compatibility-matrix-cri-o--kubernetes).
```
Comment (anubha-v-ardhan) [inline content/hi/docs/setup/production-environment/container-runtimes.md:189]:
```suggestion
अधिक जानकारी के लिए [CRI-O compatibility matrix](https://github.com/cri-o/cri-o#compatibility-matrix-cri-o--kubernetes) देखें।
```

File: content/hi/docs/setup/production-environment/container-runtimes.md [hunk 13]
```diff
@@ -0,0 +1,388 @@
...
+| ---------------- | ----------------- |
+| Debian Unstable  | `Debian_Unstable` |
+| Debian Testing   | `Debian_Testing`  |
+
+<br />
+फिर, सेट करें  `$VERSION`CRI-O संस्करण में जो आपके Kubernetes संस्करण से मेल खाता है। 
+उदाहरण के लिए, यदि आप CRI-O 1.20 इंस्टॉल करना चाहते हैं, तो `VERSION=1.20` सेट करें।
+आप अपनी स्थापना को किसी विशिष्ट रिलीज़ पर पिन कर सकते हैं।
+संस्करण 1.20.0 स्थापित करने के लिए, `VERSION=1.20:1.20.0` सेट करें।
+
```
Comment (anubha-v-ardhan) [inline content/hi/docs/setup/production-environment/container-runtimes.md:226]:
```suggestion
```

**Review by anubha-v-ardhan — CHANGES_REQUESTED** [review]
@Darshnadas Please have a look at following suggestions and try to commit suggestions directly using GitHub itself to avoid inaccuracies (like extra space, unnecessary commas and exlamation marks etc) that I believe are happening after change requests due to manually changing the page everytime
Thank you

**Comment by Darshnadas** [comment]
Okay @anubha-v-ardhan

File: content/hi/docs/setup/production-environment/container-runtimes.md (hunk 2 above)
Comment (Darshnadas) [reply content/hi/docs/setup/production-environment/container-runtimes.md:4]: Do I need to give spaces at lines 1 and 5?

File: content/hi/docs/setup/production-environment/container-runtimes.md [hunk 14]
```diff
@@ -0,0 +1,390 @@
...
+title: कंटेनर रनटाइम
+content_type: concept
+weight: 20
+---
```
Comment (anubha-v-ardhan) [inline content/hi/docs/setup/production-environment/container-runtimes.md:5]:
```suggestion
---
title: कंटेनर रनटाइम
content_type: concept
weight: 20
---
```

**Review by anubha-v-ardhan — CHANGES_REQUESTED** [review]
@Darshnadas `---` was missing which will lead to failure while building site using Hugo. (https://github.com/kubernetes/website/pull/31857/files#r815911126)

**Comment by anubha-v-ardhan** [comment]
/lgtm
/approve

Giving this PR lgtm (as discussed on Slack [here](https://kubernetes.slack.com/archives/CJ14B9BDJ/p1646132133634199?thread_ts=1646118155.299799&cid=CJ14B9BDJ)). Will make an issue to update the page with remaining fixes.

**Pull request merged**